core/src/test/scala/integration/kafka/admin/RemoteTopicCRUDTest.scala

```diff
@@ -0,0 +1,124 @@
+import org.junit.jupiter.params.provider.ValueSource
+
+import java.util.Properties
+import scala.collection.Seq
+import scala.util.Random
+import org.junit.jupiter.api.Assertions._
+import org.junit.jupiter.api.function.Executable
+
+import java.util.concurrent.ExecutionException
+
+@Tag("integration")
+class RemoteTopicCRUDTest extends KafkaServerTestHarness with Logging {
+
+  private val numPartitions = 2
+  private val numReplicationFactor = 2
+  private val defaultReplicationFactor = 1.toShort
+  private val numBrokers = 2
+
+  private var testTopicName: String = _
+  private var admin: Admin = _
+
+  /**
+   * Implementations must override this method to return a set of KafkaConfigs. This method will be invoked for every
+   * test and should not reuse previous configurations unless they select their ports randomly when servers are started.
+   */
+  override def generateConfigs: Seq[KafkaConfig] = TestUtils.createBrokerConfigs(
+    numConfigs = numBrokers,
+    zkConnect = zkConnectOrNull,
+    numPartitions = numPartitions,
+    defaultReplicationFactor = defaultReplicationFactor,
+  ).map { props =>
+    props.putAll(overrideProps())
+    KafkaConfig.fromProps(props)
+  }
+
+  override protected def kraftControllerConfigs(): Seq[Properties] = {
+    Seq(overrideProps())
+  }
+
+  @BeforeEach
+  override def setUp(info: TestInfo): Unit = {
+    super.setUp(info)
+    val props = new Properties()
+    props.put(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers())
+    admin = Admin.create(props)
+    testTopicName = s"${info.getTestMethod.get().getName}-${Random.alphanumeric.take(10).mkString}"
+  }
+
+  @AfterEach
+  def close(): Unit = {
+    if (admin != null)
+      admin.close()
+  }
+
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testTopicDeletion(quorum: String): Unit = {
+    val numPartitions = 2
+    val topicConfig = new Properties()
+    topicConfig.put(TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG, "true")
+    topicConfig.put(TopicConfig.RETENTION_MS_CONFIG, "200")
+    topicConfig.put(TopicConfig.LOCAL_LOG_RETENTION_MS_CONFIG, "100")
+    TestUtils.createTopicWithAdmin(admin, testTopicName, brokers, numPartitions, numReplicationFactor,
+      topicConfig = topicConfig)
+    TestUtils.deleteTopicWithAdmin(admin, testTopicName, brokers)
+    assertThrowsException(classOf[UnknownTopicOrPartitionException],
+      () => TestUtils.describeTopic(admin, testTopicName))
+  }
+
+  private def overrideProps(): Properties = {
+    val props = new Properties()
+    props.put(RemoteLogManagerConfig.REMOTE_LOG_STORAGE_SYSTEM_ENABLE_PROP, "true")
+    props.put(RemoteLogManagerConfig.REMOTE_STORAGE_MANAGER_CLASS_NAME_PROP,
+      classOf[NoOpRemoteStorageManager].getName)
+    props.put(RemoteLogManagerConfig.REMOTE_LOG_METADATA_MANAGER_CLASS_NAME_PROP,
+      classOf[NoOpRemoteLogMetadataManager].getName)
```

Comment:
Tried to capture the remote log segment deletion events, but couldn't extend `NoOpRemoteStorageManager.java` class:

```
class MyRemoteStorageManager extends NoOpRemoteStorageManager {
  override def deleteLogSegmentData(remoteLogSegmentMetadata: RemoteLogSegmentMetadata): Unit = {
    counter += 1;
  }
}
```  
 Caused by: java.lang.NoSuchMethodException: kafka.admin.RemoteTopicCRUDTest$MyRemoteStorageManager.<init>()
	at java.lang.Class.getConstructor0(Class.java:3082)
	at java.lang.Class.getConstructor(Class.java:1825)
	
Since, remote-storage-manager is instantiated via reflection, it expects default constructor. Tried the below ways but it still doesn't work:

1. Used companion object `Object#apply`. 
2. Added default constructor to `NoOpRemoteStorageManager`
3. Tried changing the `getDeclaredConstructor` to `getConstructor` in RemoteLogManger which uses reflection to instantiate this class.

If we want to assert the remote log segment deletion, then we need to add one more implementation for RemoteStorageManager and RemoteLogMetadataManager (similar to NoOp implementation) in scala. Let me know if it's OK to add them.